lib/server/sts.js

```diff
@@ -0,0 +1,1810 @@
+
+    if (!arnMatch) {
+        throw new errors.InvalidParameterError(
+            'Invalid RoleArn format.' +
+                ' Expected: arn:(aws|manta|triton):iam::UUID:role/role-name');
+    }
+
+    // Extract and validate components
+    //var arnPrefix = arnMatch[1];     // aws, manta, or triton
+    var accountUuid = arnMatch[2];   // UUID
+    var roleName = arnMatch[3];      // role name
+
+    // Validate UUID format is correct (additional check)
+    /* JSSTYLED */
+    var uuidPattern = /^[a-f0-9]{8}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{12}$/;
+    if (!uuidPattern.test(accountUuid)) {
+        throw new errors.InvalidParameterError(
+            'Invalid account UUID format in RoleArn');
+    }
+
+    // Validate role name (AWS-compatible character set)
+    if (!/^[a-zA-Z0-9+=,.@_-]{1,64}$/.test(roleName)) {
+        throw new errors.InvalidParameterError(
+            'Invalid role name format.' +
+                ' Must be 1-64 characters: a-zA-Z0-9+=,.@_-');
+    }
+
+    // Validate session name (AWS-compatible)
+    if (!/^[a-zA-Z0-9+=,.@_-]{2,64}$/.test(roleSessionName)) {
+        throw new errors.InvalidParameterError(
+            'Invalid RoleSessionName format. ' +
+                ' Must be 2-64 characters: a-zA-Z0-9+=,.@_-');
+    }
+
+    // Validate duration (AWS limits)
+    if (isNaN(durationSeconds) || durationSeconds < 900 ||
+        durationSeconds > 43200) {
+        throw new errors.InvalidParameterError(
+            'DurationSeconds must be between 900 and 43200');
+    }
+
+    // Enforce maximum length limits to prevent DoS
+    if (roleArn.length > 2048) {
+        throw new errors.InvalidParameterError(
+            'RoleArn too long (max 2048 characters)');
+    }
+
+    if (roleSessionName.length > 64) {
+        throw new errors.InvalidParameterError(
+            'RoleSessionName too long (max 64 characters)');
+    }
+
+    // Additional security checks for common injection patterns
+    if (roleArn.indexOf('\0') !== -1 || roleSessionName.indexOf('\0') !== -1) {
+        throw new errors.InvalidParameterError(
+            'Null bytes not allowed in parameters');
+    }
+
+    if (roleArn.indexOf('..') !== -1 || roleSessionName.indexOf('..') !== -1) {
+        throw new errors.InvalidParameterError(
+            'Path traversal patterns not allowed');
+    }
+}
+
+/**
+ * @brief Generates cryptographically secure UUID for security operations
+ *
+ * Creates a version 4 UUID using cryptographically secure random bytes
+ * for use in security-critical contexts like session tokens.
+ *
+ * @return {string} UUID in standard format (36 characters)
+ *
+ * @note Uses crypto.randomBytes() for cryptographic security
+ * @note Format: xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx
+ *
+ * @example
+ * var id = generateUUID();
+ * // Returns: "a1b2c3d4-e5f6-4789-a012-3456789abcde"
+ *
+ * @since 1.0.0
+ * @security FIXED: Replaced Math.random() with crypto.randomBytes()
```

Comment:
Not sure we need this `FIXED` comment here?

Also, there's a fair amount of usage of the [uuid](https://hound.tritondatacenter.com/?q=%22uuid%22%3A%20%223.&i=nope&files=package.json&repos=) module in other repos. It may be worth noting why we've opted not to use it here.